credsweeper/utils/util.py

```diff
@@ -108,6 +108,37 @@ def read_file(cls, path: str, encodings: Tuple[str, ...] = default_encodings) ->
                 logging.error(f"Unexpected Error: Can't read \"{path}\" as {encoding}. Error message: {exc}")
         return file_data
 
+    @classmethod
+    def decode_bytes(cls, content: bytes, encodings: Tuple[str, ...] = default_encodings) -> List[str]:
+        """Decode content using different encodings.
+
+        Try to decode bytes according to the list of encodings "encodings"
+        occurs without any exceptions. UTF-16 requires BOM
+
+        Args:
+            content: raw data that might be text
+            encodings: supported encodings
+
+        Return:
+            list of file rows in a suitable encoding from "encodings",
+            if none of the encodings match, an empty list will be returned
+
+        """
+        lines = []
+        for encoding in encodings:
+            try:
+                text = content.decode(encoding)
+                if content != text.encode(encoding):
+                    raise UnicodeError
```

Comment:
it may be better to use chardet (https://pypi.org/project/chardet/) package to detect the encoding

Reply:
I have tested the 'chardet' package (https://github.com/iuriimet/CredSweeper/tree/test_chardet).
In many cases the encoding is detected incorrectly